scripts/externalize_js.sh

```diff
@@ -45,18 +45,29 @@ if [[ ! $has_gnu_sed ]]; then
     should_continue
 fi
 
-has_local_changes=`git diff-index HEAD` && true
-if [[ $has_local_changes ]]
-then
-    abort "You have uncommitted changes in your working tree."
-fi
 
-branch=$(git rev-parse --abbrev-ref HEAD)
-if [[ $branch == 'master' ]]
-then
-    abort "You must not be on master to run this command."
+echo "Do you want this script to commit changes for you?"
+select yn in "Yes" "No"; do
+    case $yn in
+        Yes ) autocommit=true; break;;
+        No ) autocommit=false; break;;
+    esac
+done
+
+
+if [[ $autocommit == "true" ]]; then
+    has_local_changes=`git diff-index HEAD` && true
+    if [[ $has_local_changes ]]; then
+        abort "You have uncommitted changes in your working tree."
+    fi
+
+    branch=$(git rev-parse --abbrev-ref HEAD)
+    if [[ $branch == 'master' ]]; then
+        abort "You must not be on master to run this command."
```

Comment:
Good call. Could make it do a new branch for you, if that'd fit into your workflow.